### PR TITLE
Use reusable workflows over a GitHub Action

### PR DESCRIPTION
**Pull request by rogermparent**
See [the docs](https://docs.github.com/en/actions/using-workflows/reusing-workflows)

Seems like this can be a best of both worlds situation, where we can use a normal CLI build process but not need to copy GHA workflows over every site on update. The old GHA build process and subrepo shenanigans have been removed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,11 +10,9 @@ quickly ensure all new links are valid.
 
 ## Installation
 
-### CLI
-
-To get the CLI runner, install the NPM package `repo-link-check`, which contains
-a binary entry of the same name. From there, it's recommended to use the binary
-in `package.json` scripts with options for the site in question.
+Install `repo-link-check`, which contains a binary entry of the same name. From
```
Comment (casperdcl) [inline README.md:13]: install? how? npm?

File: README.md [hunk 2]
```diff
@@ -26,318 +24,3 @@ For example:
...
-The source repo can be found at
-[iterative/link-check](https://github.com/iterative/link-check). There's also a
-dist repo for the GitHub Action at
-[iterative/link-check.action](https://github.com/iterative/link-check.action),
-which serves to host only the code required to run in GitHub Actions CI and be
-posted on the marketplace.
-
-To test out a source-built CLI runner, build it with `yarn build` and then run
-`dist/cli.js` with `node` and the flags to test with in the directory of the
-project to test on.
```
Comment (casperdcl) [inline README.md:323]: what's the replacement for this deleted section?

**Comment by rogermparent** [comment]
Good points on the README, I'll follow up with an overhaul and add instructions for the reusable workflows as well.

File: .github/workflows/link-check-deployment-status.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+name: Check new links against deployment
+on:
+  workflow_call:
+    inputs:
+      main:
```
Comment (casperdcl) [inline .github/workflows/link-check-deployment-status.yml:5]: - could rename this to `branch`?
- or better, auto-detect via something like `gh repo view -R ${{ github.event.deployment.payload.repository.full_name }} --json defaultBranchRef --jq .defaultBranchRef.name` (untested)

Comment (rogermparent) [reply .github/workflows/link-check-deployment-status.yml:5]: I'll have to poke around in the data structure again, there's got to be some way we can find the main branch even if it takes an API call or two, but maybe even without one.

Comment (rogermparent) [reply .github/workflows/link-check-deployment-status.yml:5]: Found it in an old context dump, `github.event.repository.default_branch` is what we want.

Comment (rogermparent) [reply .github/workflows/link-check-deployment-status.yml:5]: https://github.com/iterative/link-check/pull/18 and https://github.com/iterative/cml.dev/pull/287 are now opened for this!

**Pull request merged**